float32 angle
---
bool success
